File: frontend/src/api/watchlistService.js
//src/api/watchlistService.js

import axiosInstance from './axiosConfig';

// Get the user's watchlist
export const getWatchlist = () => {
  return axiosInstance.get('/watchlist');
};

// Add a ticker to the watchlist
export const addToWatchlist = (ticker) => {
  return axiosInstance.post('/watchlist/add', { ticker });
};

// Remove a ticker from the watchlist
export const removeFromWatchlist = (ticker) => {
  return axiosInstance.delete(`/watchlist/remove/${ticker}`);
};

File: investify-backend/src/main/java/com/investify/investify_backend/service/WatchlistService.java
package com.investify.investify_backend.service;

import com.investify.investify_backend.entity.Watchlist;
import java.util.List;

public interface WatchlistService {
    List<Watchlist> getWatchlist(String userEmail);
    Watchlist addToWatchlist(String userEmail, String ticker);
    void removeFromWatchlist(String userEmail, String ticker);
}